src/software/ai/evaluation/robot.h

```diff
@@ -28,26 +28,45 @@ bool robotOrientationWithinAngleThresholdOfTarget(Point position, Angle orientat
  * Determines if a robot has possession of the ball. A robot is considered to have
  * possession if the ball is in a area close to its dribbler.
  *
- * @param  The ball that is wanted to be possessed
- * @param robot The Robot which wants to know if it has the ball.
+ * @param ball_states The circular buffer of TimestampedBallState
+ * @param robot_states The circular buffer of TimestampedRobotState
  * @param timestamp The time at which we want to know if the robot had the ball
  * @return True if the ball is close to the front dribbler, False if the ball is not,
  *          and nullopt if we don't have information for the given timestamp. This
  * function is guaranteed to return non-nullopt if no timestamp is passed in.
  */
-std::optional<bool> robotHasPossession(const Ball& ball, const Robot& robot,
-                                       std::optional<Timestamp> timestamp = std::nullopt);
+std::optional<bool> robotHasPossession(
+    const boost::circular_buffer<TimestampedBallState>& ball_states,
+    const boost::circular_buffer<TimestampedRobotState>& robot_states,
+    std::optional<Timestamp> timestamp = std::nullopt);
```

Comment:
This will make it compatible with sensorfusion in the future because sensorfusion has circular buffers

Reply:
In that case, IMO this change really should come when we migrate these functions into sensorfusion. 

@akhilveeraghanta @MathewMacDougall for your thoughts

Reply:
No strong opinion. This change still matches the theme of this PR by simplifying the interface to just the buffers of data it needs, rather than full `Ball` and `Robot` types

Reply:
So I still disagree with this change, but if we're going to move this to sensorfusion in the immediate future then I won't hold this PR up.